oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilter.java

```diff
@@ -251,11 +252,13 @@ private void sendAuthorizationConsent(HttpServletRequest request, HttpServletRes
 		String state = authorizationConsentAuthentication.getState();
 
 		if (hasConsentUri()) {
-			String redirectUri = UriComponentsBuilder.fromUriString(resolveConsentUri(request))
```

Comment:
This needs to be reverted as per previous comment. No changes are necessary in `sendAuthorizationConsent()` since the `state` parameter is generated internally using a `Base64.getUrlEncoder()`. Only apply changes in `sendAuthorizationResponse()`.